### PR TITLE
feat: create knockout mode

### PR DESCRIPTION
**Pull request by nguyenquan23**

### CODE REVIEW

File: src/main/java/com/example/thetelephoneappbe/controller/UserController.java [hunk 1]
```diff
@@ -48,83 +59,119 @@ public void setRoleService(RoleService roleService) {
...
-                return  ResponseEntity.ok("nick name is duplicate");
-            }
+    @PostMapping(ApiJoinRoom.API_JOIN_ROOM)
+    public ResponseEntity<String> joinRoom(@PathVariable(ApiJoinRoom.PARAM_ID_ROOM) Long idRoom,
+                                           @PathVariable(ApiJoinRoom.PARAM_USER_NAME) String userName,
+                                           @PathVariable(ApiJoinRoom.PARAM_ID_AVATAR) String id_avatar) {
 
+      try {
+          Room roomPlay = roomService.getOneRoom(idRoom);
+        if (roomPlay.getUsers().size() < roomPlay.getMaxPlayer()) {
```
Comment (ces-khanhhoang) [inline src/main/java/com/example/thetelephoneappbe/controller/UserController.java:84]: add comment logic for this code block

File: src/main/java/com/example/thetelephoneappbe/controller/UserController.java [hunk 2]
```diff
@@ -167,12 +218,14 @@ public ResponseEntity<String> Done(@PathVariable("id_room") Long idRoom, @PathVa
...
-    public ResponseEntity<String> result(@PathVariable("nickname") String nickname, @PathVariable("id_room") Long idRoom) {
+    @PostMapping(ApiGetResult.API_GET_RESULT)
+    public ResponseEntity<String> getResult(@PathVariable(ApiGetResult.PARAM_NICKNAME) String nickname,
+                                            @PathVariable(ApiGetResult.PARAM_ID_ROOM) Long idRoom) {
         StorageGame storageGamePlay = storageGames.stream()
                 .filter(storageGame -> storageGame.getIdRoom().equals(idRoom))
                 .reduce((first, second) -> second).orElseThrow();
-        int count = 0;
+
+
```
Comment (ces-khanhhoang) [inline src/main/java/com/example/thetelephoneappbe/controller/UserController.java]: remove one empty line

File: src/main/java/com/example/thetelephoneappbe/controller/UserController.java [hunk 3]
```diff
@@ -182,34 +235,38 @@ public ResponseEntity<String> result(@PathVariable("nickname") String nickname,
         Map<Integer, String> data = storageGamePlay.getResult().get(nickname);
 
         for (Map.Entry<Integer, String> entry : data.entrySet()) {
-            ResultDTO resultDTO = new ResultDTO(nicks.get(count),entry.getValue(),avatars.get(count));
+            System.out.println(entry.getKey());
```
Comment (ces-khanhhoang) [inline src/main/java/com/example/thetelephoneappbe/controller/UserController.java]: remove this

File: src/main/java/com/example/thetelephoneappbe/controller/UserController.java [hunk 4]
```diff
@@ -182,34 +235,38 @@ public ResponseEntity<String> result(@PathVariable("nickname") String nickname,
...
 
     }
 
-    @PostMapping("/play/{number}/{id_room}")
-    public  ResponseEntity<String> numberPlay(@PathVariable("id_room") Long idRoom,@PathVariable("number") Integer number ){
-        Room playRoom =  roomService.getOneRoom(idRoom);
-        playRoom.setStatus("MAX");
+    @PostMapping(ApiChooseMaxNumber.API_CHOOSE_MAX_NUMBER)
+    public  ResponseEntity<String> numberPlay(@PathVariable(ApiChooseMaxNumber.PARAM_ID_ROOM) Long idRoom,
+                                              @PathVariable(ApiChooseMaxNumber.PARAM_NUMBER) Integer number ) {
```
Comment (ces-khanhhoang) [inline src/main/java/com/example/thetelephoneappbe/controller/UserController.java]: remove the white space behind number

**Review by ces-khanhhoang — APPROVED** [review]

**Review by noonoo169 — APPROVED** [review]

**Pull request merged**